app/components/channel_drawer/channels_list/channel_unread_item/index.js

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {getCurrentChannelId, getChannel, getMyChannelMember} from 'mattermost-redux/selectors/entities/channels';
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import ChannelUnreadItem from './channel_unread_item';
+
+function mapStateToProps(state, ownProps) {
+    const channel = getChannel(state, ownProps.channelId);
+    const member = getMyChannelMember(state, ownProps.channelId);
+
+    return {
+        currentChannelId: getCurrentChannelId(state),
+        displayName: channel.display_name,
+        hasUnread: channel.total_msg_count - member.msg_count > 0,
```

Comment:
Isn't this always true for unread channels?

Reply:
doh!